### PR TITLE
Fix types for sessions

### PR DESCRIPTION
**Pull request by ljodal**
The session key can be None, which type stubs no longer allow after #1129, and the session_dict can be `dict[str, Any]`

### CODE REVIEW

File: django-stubs/contrib/sessions/backends/base.pyi [hunk 1]
```diff
@@ -24,14 +24,14 @@ class SessionBase(Dict[str, Any]):
     def items(self): ...
     def clear(self) -> None: ...
     def is_empty(self) -> bool: ...
-    def _get_session_key(self) -> str: ...
+    def _get_session_key(self) -> Optional[str]: ...
     def _set_session_key(self, value: str) -> None: ...
```
Comment (ljodal) [inline django-stubs/contrib/sessions/backends/base.pyi]:
```suggestion
    def _set_session_key(self, value: Optional[str]) -> None: ...
```

**Comment by ljodal** [comment]
I found some problems with `session_dict` as well in `BaseSession`, so I fixed those as well

**Review by adamchainz — APPROVED** [review]
LGTM!

**Review by sobolevn — COMMENTED** [review]
Thank you! Can you please run `black` to fix https://github.com/typeddjango/django-stubs/actions/runs/3385096206/jobs/5622845846 ?

**Comment by sobolevn** [comment]
I think I can enable `pre-commit.ci` to fix this automatically 🤔

**Comment by adamchainz** [comment]
Yes pre-commit.ci ftw!

**Comment by adamchainz** [comment]
I guess my local pre-commit didn't get applied because this was a rebase commit, and git doesn't fire hooks during rebase 😢

**Pull request merged**